rtsp.go

```diff
@@ -399,7 +470,95 @@ func (rc *rtspCamera) initMJPEG(tracks media.Medias, baseURL *url.URL) error {
 
 		rc.latestFrame.Store(&img)
 	})
+	return nil
+}
+
+// SubscribeRTP registers the PacketCallback which will be called when there are new packets.
+// NOTE: Packets may be dropped before calling packetsCB if the rate new packets are received by
+// the rtppassthrough.Source is greater than the rate the subscriber consumes them.
+
+func (rc *rtspCamera) SubscribeRTP(ctx context.Context, bufferSize int, packetsCB rtppassthrough.PacketCallback) (rtppassthrough.SubscriptionID, error) {
+	if !rc.rtpH264Passthrough {
+		return uuid.Nil, ErrH264PassthroughNotEnabled
+	}
+
+	sub, err := rtppassthrough.NewStreamSubscription(bufferSize, func(err error) { rc.logger.Errorw("stream subscription hit error", "err", err) })
+	if err != nil {
+		return uuid.Nil, err
+	}
+	webrtcPayloadMaxSize := 1188 // 1200 - 12 (RTP header)
+	encoder := &rtph264.Encoder{
+		PayloadType:    96,
+		PayloadMaxSize: webrtcPayloadMaxSize,
+	}
+
+	if err := encoder.Init(); err != nil {
+		return uuid.Nil, err
+	}
+
+	var firstReceived bool
+	var lastPTS time.Duration
+	// OnPacketRTP will call this unitSubscriberFunc for all subscribers.
+	// unitSubscriberFunc will then convert the Unit into a slice of
+	// WebRTC compliant RTP packets & call packetsCB, which will
+	// allow the caller of SubscribeRTP to handle the packets.
+	// This is intended to free the SubscribeRTP caller from needing
+	// to care about how to transform RTSP compliant RTP packets into
+	// WebRTC compliant RTP packets.
+	unitSubscriberFunc := func(u unit.Unit) error {
+		tunit, ok := u.(*unit.H264)
+		if !ok {
+			return errors.New("(*unit.H264) type conversion error")
+		}
+
+		// If we have no AUs we can't encode packets.
+		if tunit.AU == nil {
+			return nil
+		}
+
+		if !firstReceived {
+			firstReceived = true
+		} else if tunit.PTS < lastPTS {
+			return errors.New("WebRTC doesn't support H264 streams with B-frames")
```

Comment:
What is expected to happen if h264 feed uses b frames?

Reply:
Curious to try this out with the input camera using different libx264 b frame strategies. Such as:

```
-vcodec libx264 -pix_fmt yuv420p -bf 3 -b_strategy 2 
``

Reply:
This is inspired by https://github.com/bluenviron/mediamtx/blob/main/internal/servers/webrtc/session.go#L185 (I've commented the code as such)

Right now it will just kill the video stream subscription & the caller won't get packets if the H264 stream has B-frames. 

In the future, when I get https://github.com/viamrobotics/rdk/pull/3792 working, I intend to change the signature of camera.SubscribeRTP such that the caller can detect when the subscription dies (such as happens in this case) & act accordingly, by, for example, falling back to GetImage.